modules/aws/sonar-upgrader/python_upgrader/upgrade/main.py

```diff
@@ -415,79 +417,63 @@ def collect_python_location(extended_node, stop_on_failure, upgrade_status_servi
         return None
 
 
-def run_all_preflight_validations(agentless_gw_extended_node_dict, dsf_hub_extended_node_dict, target_version,
-                                  stop_on_failure, upgrade_status_service):
+def run_preflight_validations(args, agentless_gw_extended_node_dict,
+                                  dsf_hub_extended_node_dict, upgrade_status_service):
     print("----- Preflight validations")
 
-    gws_preflight_validations_passed = run_preflight_validations_for_extended_nodes(agentless_gw_extended_node_dict,
-                                                                                    target_version,
-                                                                                    PREFLIGHT_VALIDATIONS_SCRIPT_NAME,
-                                                                                    stop_on_failure,
-                                                                                    upgrade_status_service)
-    hub_preflight_validations_passed = run_preflight_validations_for_extended_nodes(dsf_hub_extended_node_dict,
-                                                                                    target_version,
-                                                                                    PREFLIGHT_VALIDATIONS_SCRIPT_NAME,
-                                                                                    stop_on_failure,
-                                                                                    upgrade_status_service)
-    return gws_preflight_validations_passed and hub_preflight_validations_passed
-
-
-def run_preflight_validations_for_extended_nodes(extended_node_dict, target_version, script_file_name, stop_on_failure,
-                                                 upgrade_status_service):
-    all_success_or_skip = True
-    for extended_node in extended_node_dict.values():
-        success_or_skip = maybe_run_preflight_validations_for_extended_node(extended_node, target_version,
-                                                                            script_file_name, stop_on_failure,
-                                                                            upgrade_status_service)
-        all_success_or_skip = all_success_or_skip and success_or_skip
-    return all_success_or_skip
-
+    successful = True
+    for node in chain(agentless_gw_extended_node_dict.values(), dsf_hub_extended_node_dict.values()):
+        if upgrade_status_service.should_run_preflight_validations(node.get('dsf_node_id')):
+            successful = successful and run_preflight_validations_for_node(args, node, upgrade_status_service)
```

Comment:
4. The main issue - the change in function run_preflight_validations (see comment in the code) broke the automation when stop_on_failure = false. In this case, if one of the nodes' validations failed, it won't continue to the next nodes.

Reply:
Short circuit issue has been fixed.